dynamiqs/solvers/result.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from torch import Tensor
+
+from ..options import Options
+
+
+def memory_bytes(x: Tensor) -> int:
+    return x.element_size() * x.numel()
+
+
+def memory_str(x: Tensor) -> str:
+    mem = memory_bytes(x)
+    if mem < 1024**2:
+        return f'{mem / 1024:.2f} Kb'
+    elif mem < 1024**3:
+        return f'{mem / 1024**2:.2f} Mb'
+
+    return f'{mem / 1024**3:.2f} Gb'
+
+
+def tensor_str(x: Tensor) -> str:
+    return f'tensor {tuple(x.size())} [{memory_str(x)}]'
```

Comment:
```suggestion
    return f'Tensor {tuple(x.size())} | {memory_str(x)}'
```
Just a suggestion, I like it a tiny bit more:
```python
y_save     : Tensor (4, 101, 64, 64) | 12.62 Mb 
exp_save   : Tensor (4, 2, 101) | 6.31 Kb
```